03_python_fastapi_project/main.py

```
@@ -111,6 +112,126 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
+        result = await db.execute(
+            select(BasketItem).options(selectinload(BasketItem.product))
+            .filter(BasketItem.id == db_item.id)
+        )
+        return result.scalar_one()
+
+
+@app.put("/basket/{item_id}", response_model=BasketItemDTO)
+async def update_basket_item(
+    item_id: int,
+    item_update: BasketItemUpdate,
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.13.3)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)